rfc/2020/alertmanager-integration/christyjacob4.md

```diff
@@ -0,0 +1,503 @@
+      - configMap:
+          name: linkerd-{{.Values.alertmanager.name}}-config
+        name: {{.Values.alertmanager.name}}-config
+      - {{- include "partials.proxy.volumes.identity" . | indent 8 | trimPrefix (repeat 7 " ") }}
+
+```
+The `initContianers` is added here in a way similar to other *Linkerd* Components that configures iptables to forward the traffic correctly.
+
+Since the Alertmanager config can contain sensitive information like passwords for different channels, the proposal is to use [Kubernetes Secrets](https://kubernetes.io/docs/concepts/configuration/secret/) to store the ConfigMap. 
+
+Apart from this, the roles need to be defined for Alertmanager which is done using an RBAC file 
+
+```yaml
+---
+###
+### linkerd-alertmanager RBAC
+###
+---
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: linkerd-{{.Values.global.namespace}}-{{.Values.alertmanager.name}}
+  labels:
+    {{.Values.global.controllerComponentLabel}}: {{.Values.alertmanager.name}}
+    {{.Values.global.controllerNamespaceLabel}}: {{.Values.global.namespace}}
+rules:
+- apiGroups: [""]
+  resources: ["nodes", "nodes/proxy", "pods"]
+  verbs: ["get", "list", "watch"]
+---
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: linkerd-{{.Values.global.namespace}}-{{.Values.alertmanager.name}}
+  labels:
+    {{.Values.global.controllerComponentLabel}}: {{.Values.alertmanager.name}}
+    {{.Values.global.controllerNamespaceLabel}}: {{.Values.global.namespace}}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: linkerd-{{.Values.global.namespace}}-{{.Values.alertmanager.name}}
+subjects:
+- kind: ServiceAccount
+  name: linkerd-{{.Values.alertmanager.name}}
+  namespace: {{.Values.global.namespace}}
+---
+kind: ServiceAccount
+apiVersion: v1
+metadata:
+  name: linkerd-{{.Values.alertmanager.name}}
+  namespace: {{.Values.global.namespace}}
+  labels:
+    {{.Values.global.controllerComponentLabel}}: {{.Values.alertmanager.name}}
+    {{.Values.global.controllerNamespaceLabel}}: {{.Values.global.namespace}}
+```
+
+---
+- ## Service Profile Integration 
+
+Service Profiles is an extremely useful *Linkerd* Feature that enables a developer to get metrics on a per route basis. A Service profile can be generated by *Linkerd* using the `linkerd profile` command and passed onto `kubectl apply`. 
+
+`linkerd profile` can obtain route information in 4 different ways:
+- Swagger
+- Protobuf
+- Auto-creation
+- Template
+
+Hence based on the availability of either of these specifications, we can derive per route metrics and create default Prometheus rules for the same. The `config.yaml` file will allow the specification of either a swagger file or a protobuf file. 
+```yaml
+publicWebhook: false
+alertmanager:
+  config: path/to/config/file.yaml 
+
+prometheus:
+  rules: path/to/rules/file.yaml
+  swaggerSpecPath: path/to/swagger/spec
+  protobufSpecPath: path/to/protobuf/spec
```

Comment:
Are you planning on creating the service profiles resources based on these files? What if the users already created them?